### PR TITLE
The great gas expansion

### PR DESCRIPTION
**Pull request by dwasint**
# About The Pull Request
Adds a ton of new gases, aswell as methods of obtaining gases
adds cold fusion

adds freon, healium, hydrogen, pluonium, zauker. halon, hexane and dilthium gases
changes fusion to use hydrogen instead of co2
added fusion interactions for both co2 and h2 when its active
gave water sm properties (12 heat modifier, -2 transmit modifier)
heat exchange can now lower the sm down to 3k down from 27k
crystalizer added with recipes for it
electrolyzer added to produce hydrogen

https://github.com/yogstation13/Yogstation/pull/5847
https://github.com/tgstation/tgstation/pull/49821
https://github.com/yogstation13/Yogstation/pull/10447

## Why It's Good For The Game
More stuff for atmos to do, the ability for atmos to get some actual gamer gear

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog

:cl:
add: freon, hydrogen, healium, dilithium, pluonium, halon, zauker, hexane as new gases
add: crystalizer you can now solidify gases into cool objects
add: metalic hydrogen, go forth atmosians and make an army armed with fireaxes
add: electrolyzer, used to make new gases check the tg wiki for a better idea
add: cold fusion
tweak: heat generation of some gases
change: fusion uses hydrogen instead of co2
add: gave water sm properties (12 heat modifier, -2 transmit modifier)
change: heat exchange can now lower the sm down to 3k down from 27k
add: added fusion interactions for both co2 and h2 when its active
/:cl:

### CODE REVIEW

**Comment by DoctorSquishy** [comment]
https://www.youtube.com/watch?v=atuFSv2bLa8

**Comment by dwasint** [comment]
i want this enough to work on it again

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 1]
```diff
@@ -611,3 +663,398 @@
...
+	air.adjust_moles(GAS_STIMULUM, -(heat_efficency * 0.5))
+	air.adjust_moles(GAS_ZAUKER, (heat_efficency * 0.5))
+
+	if(energy_used)
+		var/new_heat_capacity = air.heat_capacity()
+		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
+			air.set_temperature(max(((old_thermal_energy - energy_used) / new_heat_capacity), TCMB))
+	return REACTING
+
+/datum/gas_reaction/halon_o2removal
```
Comment (nednaZ) [inline code/modules/atmospherics/gasmixtures/reactions.dm:900]: Looks like they forgot an underscore
```suggestion
/datum/gas_reaction/halon_o2_removal
```

**Review by nednaZ — CHANGES_REQUESTED** [review]
This will be going into the next testmerge after the changes are made

File: monkestation/code/datums/components/hot_ice.dm [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	src.gas_name = gas_name
+	src.gas_amount = gas_amount
+	src.temp_amount = temp_amount
+
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/attackby_react)
+	RegisterSignal(parent, COMSIG_ATOM_FIRE_ACT, .proc/flame_react)
+
+/datum/component/thermite/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_PARENT_ATTACKBY)
+	UnregisterSignal(parent, COMSIG_ATOM_FIRE_ACT)
```
[the remark this answers is not in the document]
Comment (dwasint) [reply monkestation/code/datums/components/hot_ice.dm:17]: forgor to rename the unregister

File: code/game/objects/items/stacks/bscrystal.dm [hunk 3]
```diff
@@ -98,3 +98,26 @@
 			to_chat(user, "<span class='notice'>You break off a crystal.</span>")
 	else
 		..()
+
+//Dilithium crystals, used in atmospherics to lower the required temperature for fusion, by grinding into a matter and reacting to form a gas.
+/obj/item/stack/ore/dilithium_crystal
```
[the remark this answers is not in the document]
Comment (dwasint) [reply code/game/objects/items/stacks/bscrystal.dm:103]: our orm ignores it

[the remark this answers is not in the document]
Comment (dwasint) [reply code/game/objects/items/stacks/bscrystal.dm:103]: so do lathes

**Comment by nednaZ** [comment]
Been getting reports of fusion not working in general, even from people that know it.

**Review by nednaZ — APPROVED** [review]
This apparently works and I haven't been seeing any issues.
Atmospherics will forever be a mystery, even if I read and understand the code.

**Comment by Ghilker** [comment]
You could've at least posted the prs you're taking most of these from

**Pull request merged**